testing-common/integration-tests/src/main/java/context/ContextTestInstrumentation.java

```diff
@@ -42,78 +44,71 @@ public void transform(TypeTransformer transformer) {
 
   @SuppressWarnings("unused")
   public static class MarkInstrumentedAdvice {
+    @AssignReturned.ToReturned
     @Advice.OnMethodExit
-    public static void methodExit(@Advice.Return(readOnly = false) boolean isInstrumented) {
-      isInstrumented = true;
+    public static boolean methodExit() {
+      return true;
     }
   }
 
   @SuppressWarnings("unused")
   public static class StoreAndIncrementApiUsageAdvice {
+    @AssignReturned.ToReturned
     @Advice.OnMethodExit
-    public static void methodExit(
-        @Advice.This KeyClass thiz, @Advice.Return(readOnly = false) int contextCount) {
-
-      VirtualField<KeyClass, Context> virtualField =
-          VirtualField.find(KeyClass.class, Context.class);
-
-      Context context = virtualField.get(thiz);
+    public static int methodExit(@Advice.This KeyClass thiz) {
+      Context context = CONTEXT.get(thiz);
       if (context == null) {
         context = new Context();
-        virtualField.set(thiz, context);
+        CONTEXT.set(thiz, context);
       }
 
-      contextCount = ++context.count;
+      return ++context.count;
     }
   }
 
   @SuppressWarnings("unused")
   public static class GetApiUsageAdvice {
+
+    @Advice.OnMethodEnter(skipOn = Advice.OnNonDefaultValue.class)
+    public static boolean methodEnter() {
+      // always skip original method body
+      return true;
+    }
```

Comment:
[for reviewer] Without this the original return value `-1` is returned in the `context.FieldBackedImplementationTest#removeInstanceState` test, even if the exit advice that should alter the return value executes properly.